simcase-game/src/views/Ranking.js

```diff
@@ -69,13 +65,165 @@ const button = {
   width: "50%"
 };
 
+const rankNum = {
+  fontSize: "1.2em",
+  fontWeight: "bold"
+};
+
 function Ranking() {
-  const players = leaderboardDummyData;
+  const [data, setData] = React.useState([]);
+  const [score, setScore] = React.useState(0);
+  const [user, setUser] = React.useState("");
+  const [position, setPosition] = React.useState(0);
+  const [scoreboard, setScoreboard] = React.useState(true);
+
+  React.useState(() => {
+    fire();
+  }, [data]);
+
+  function fire() {
+    firebase
+      .firestore()
+      .collection("data")
+      .orderBy("points", "desc")
+      .get()
+      .then(querySnapshot => {
+        let data = [];
+        let score = ~~(Math.random() * (1000 - 0) + 0);
+        setScore(score);
+        querySnapshot.forEach(doc => {
+          data.push({
+            username: doc.data().username,
+            points: doc.data().points
+          });
+        });
+        data.push({ username: "You", points: score });
+        data.sort((a, b) => {
+          return b.points - a.points;
+        });
+        setData(data);
+        data.filter((value, index) => {
+          if (value.points === score) {
+            setPosition(index + 1);
+          }
+        });
+      });
+  }
+
+  function addUser() {
+    firebase
+      .firestore()
+      .collection("data")
+      .add({
```

Comment:
The method "add" is available when you are using firebase right?

Reply:
Yes, it is. You can read more about it [here](https://firebase.google.com/docs/firestore/manage-data/add-data)